flink/v1.13/flink/src/main/java/org/apache/iceberg/flink/source/Position.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink.source;
+
+import java.io.Serializable;
+import java.util.Objects;
+import org.apache.flink.annotation.Internal;
+import org.apache.iceberg.CombinedScanTask;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+
+/**
+ * A mutable class that defines the read position
+ * <ul>
+ *   <li>file offset in the list of files in a {@link CombinedScanTask}</li>
+ *   <li>record offset within a file</li>
+ * </ul>
+ */
+@Internal
+public class Position implements Serializable {
+
+  private static final long serialVersionUID = 1L;
+
+  private int fileOffset;
+  private long recordOffset;
+
+  public Position(int fileOffset, long recordOffset) {
+    this.fileOffset = fileOffset;
+    this.recordOffset = recordOffset;
+  }
+
+  void advanceFile() {
+    this.fileOffset += 1;
+    this.recordOffset = 0L;
+  }
+
+  void advanceRecord() {
+    this.recordOffset += 1L;
+  }
+
+  public void update(int newFileOffset, long newRecordOffset) {
```

Comment:
Seems more like a `set` method to me since it directly sets the internal counters rather than advancing them using an amount.

Reply:
sure. will rename the method to `set`